### PR TITLE
Small typo fixes (mostly in AIEOps.td)

### PR DESCRIPTION
**Pull request by hunhoffe**
I was reading through `AIEOps.td` and figured I may as well fix any typos I happened to notice.

The majority of changes in this PR are to change `dimensionsFromStream` and `dimensionsToStream` to use a consistent name throughout different levels of code (in parser, in op attribute name, in python, etc.). This PR does not make the naming completely consistent, as I left a `dimensionsFromStream` attribute whereas different areas of code (and in the python) use `dimensionsFromStreamPerConsumer`. I kept the two names because `dimensionsFromStream` is used per-consumer (an element in a list) whereas `dimensionsFromStreamPerConsumer` typically refers to a list of dimensions to stream. However, if it would be better to make it all consistent, I can make that modification.

### CODE REVIEW

File: include/aie/Dialect/AIE/IR/AIEOps.td [hunk 1]
```diff
@@ -1650,15 +1650,15 @@ def AIE_ObjectFifoCreateOp: AIE_Op<"objectfifo", [HasParent<"DeviceOp">, Symbol]
 
     The following example shows an objectFifo which transposes a 16x16 matrix of
     `i32`s on the producer side using strides and sizes. No transformation is
-    applied on the consumer side of `%tile13` (in this case the `fromStream`
+    applied on the consumer side of `%tile13` (in this case the `dimensionsFromStreamPerConsumer`
     attribute may also be left off), and a transformation on `%tile23` first gives
```
Comment (hunhoffe) [inline include/aie/Dialect/AIE/IR/AIEOps.td:1654]: @abisca It seems like the example below doesn't match the description (e.g., fromStream is not left off) or maybe I'm reading it wrong. Can you help correct me in understanding what this examples is showing and how to describe it?

File: include/aie/Dialect/AIE/IR/AIEOps.td [hunk 2]
```diff
@@ -1650,15 +1650,15 @@ def AIE_ObjectFifoCreateOp: AIE_Op<"objectfifo", [HasParent<"DeviceOp">, Symbol]
...
     all even indices from the stream, followed by all odd indices:
 
     ```
-      aie.objectfifo @of4 (%tile12 toStream [<16, 1>, <16, 16>, <1,1>],
+      aie.objectfifo @of4 (%tile12 dimensionsToStream [<16, 1>, <16, 16>, <1,1>],
```
Comment (AndraBisca) [inline include/aie/Dialect/AIE/IR/AIEOps.td:1658]: The original keyword "toStream" was correct: you can see the custom parser for it [here](https://github.com/Xilinx/mlir-aie/blob/main/lib/Dialect/AIE/IR/AIEDialect.cpp#L517).

File: include/aie/Dialect/AIE/IR/AIEOps.td (hunk 1 above)
Comment (AndraBisca) [reply include/aie/Dialect/AIE/IR/AIEOps.td:1654]: The example shows that there is no data layout transformation on the consumer %tile13, you can see that the fromStream attribute there is `[]`. This is different for consumer %tile23 which has a data layout transformation expressed as `[<2, 1>, <128, 2>]`

**Review by AndraBisca — APPROVED** [review]
Thank you Erika for all of these changes! Except for my few remarks I think the PR looks good! :)

File: include/aie/Dialect/AIE/IR/AIEOps.td (hunk 1 above)
Comment (hunhoffe) [reply include/aie/Dialect/AIE/IR/AIEOps.td]: Thank you, that was my mistake to read it incorrectly!

File: include/aie/Dialect/AIE/IR/AIEOps.td (hunk 2 above)
Comment (hunhoffe) [reply include/aie/Dialect/AIE/IR/AIEOps.td:1658]: For consistency, as discussed offline, I updated the parser

File: include/aie/Dialect/AIE/IR/AIEOps.td [hunk 3]
```diff
@@ -1650,15 +1650,15 @@ def AIE_ObjectFifoCreateOp: AIE_Op<"objectfifo", [HasParent<"DeviceOp">, Symbol]
...
                            {
-                            %tile13 fromStream [],
-                            %tile23 fromStream [<2, 1>, <128, 2>]
+                            %tile13 dimensionsFromStreamPerConsumer [],
+                            %tile23 dimensionsFromStreamPerConsumer [<2, 1>, <128, 2>]
```
[the remark this answers is not in the document]
Comment (hunhoffe) [reply include/aie/Dialect/AIE/IR/AIEOps.td]: For consistency, as discussed offline, I updated the parser

**Review by AndraBisca — APPROVED** [review]
Looks good to me!

**Review by keryell — APPROVED** [review]
Thanks for the cleaning and clarification.

**Pull request merged**